### ISSUE TITLE
Clearly support data binding for multiple select

### ISSUE THREAD

**Issue by lonerzzz** [ng-select/ng-select#1640]
**Is your feature request related to a problem? Please describe.**
In trying to use the multiselect capabilities of the selector, the examples (https://ng-select.github.io/ng-select#/multiselect) all avoid the use of the bindValue in contrast to the single select options provided for the data binding documentation (https://ng-select.github.io/ng-select#/bindings).

When I attempt to use objects and data binding for multiselect, the error "Binding object() with bindValue is not allowed." gets raised.

**Describe the solution you'd like**
For consistency, better usability of the code and avoiding special external handling, data binding should be equally documented for single and multiselect.

Right now, the documentation does not highlight the limitations/requirements of use for data binding so users don't know there is a problem until they are using the library. Likewise the multiselect options do not call out that multiselect with objects is supported only with a compareWith implementation.

**Describe alternatives you've considered**
N/A

**Comment by varnastadeus** [comment]
there is no distinction between multi or single select with this case, you will get that error either way

**Comment by lonerzzz** [comment]
I was able to use bindValue on a single select without raising this error. However, the use of bindValue on the multiselect ended up with issues. I am not sure of the reason why. That said, showing an example of data binding in the documentation with multiselect would make things clearer. I also submitted a pull request to better log how the compareWith is required https://github.com/ng-select/ng-select/pull/1639 when using bindValue.
